Showing off armor now in addition to weapons

philipmat opened the pull request:
This PR introduces a few more item specifications:
-  `head | helmet`
- `chest`
- `arms | gloves | gauntlets`
- `legs | boots | greaves`
- `class | mark | bond | cape | cloak` 
- and even `ghost`
  
  It looks like this:
  ![screen shot 2016-10-19 at 12 03 20 pm](https://cloud.githubusercontent.com/assets/709928/19529067/175eb6ac-95f4-11e6-97db-016f6fc48da8.png)

The patch depends on the weapon stats patch.

phillipspc commented:
So this one is probably better left in a forked repo. Limiting the bot to only weapons was intentional for its original purposes (used in a channel dedicated to weapon discussion). I've considered putting armor back in but decided against it because:
1. fewer people care about showing off their armor
2. higher potential for spammy usage ("hey check out my tier 12 loadout!" _calls bot 6 times_)

philipmat commented:
How would you feel about adding some config options, either via a config file or via environmental options, that would allow one to turn on/of specific settings, eg: `DISPLAY=weapons,armor,details`?

phillipspc commented:
@philipmat Thats a great thought! That would let us expand the functionality while still allowing us to limit its usage where needed. We'll just want to make sure the bot operates consistently according to this setting (ie. showing the right info in the help command)

philipmat commented:
Would you be okay with the default value being _all_? I think this would broaden the appeal of the bot and still allow to easily configure the robot for more restrictive uses such as the Crucible Radio case.

phillipspc commented:
Yeah I agree with that

philipmat commented:
Updated with a `SLOT_DISPLAY` setting. I was going for DISPLAY originally, but that's a typical 
*nix setting so I chose a more obvious one instead. 

I've updated the README as well to reflect this new option.

phillipspc commented:
Looks good @philipmat, I'll just want to test things out before merging

phillipspc commented on scripts/bungie-constants.coffee:
```diff
@@ -0,0 +1,47 @@
+module.exports = 
```
good call breaking this stuff out 👍

phillipspc commented on scripts/bungie.coffee:
```diff
@@ -195,8 +233,8 @@ getCharacterId = (bot, membershipType, playerId, robot) ->
-# Gets itemInstanceId from Inventory Summary based on weaponSlot
-getItemIdFromSummary = (bot, membershipType, playerId, characterId, weaponSlot) ->
+# Gets itemInstanceId from Inventory Summary based on bucket
+getItemIdFromSummary = (bot, membershipType, playerId, characterId, bucket, robot) ->
```
any reason we're passing in `robot` now?

philipmat replied:
Because the code was attempting to send a message to the room if anything was wrong, but there was no `robot` passed to that method:

```
    item = items.filter(matchesBucketHash)
    if item.length is 0
      robot.send {room: bot.message.user.name, "unfurl_media": false}, "Something went wrong, couldn't find the requested item for this character. #{helpText}"
```

phillipspc replied:
cool. good catch

phillipspc commented:
@philipmat sorry I let this sit for so long. Armor works great. Thanks
